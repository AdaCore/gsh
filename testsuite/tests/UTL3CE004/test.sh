echo "when 'a' does not exist but 'a.exe' does: cp a copy_a"
cp a copy_a
echo "* ls copy_a.exe"
ls copy_a.exe
echo "* cat copy_a.exe"
cat copy_a.exe

rm -r copy_a.exe

echo ""
echo "copy on side of an existing file: cp a b"
cp a b
echo "* ls b*"
ls b*
echo "* cat b*"
cat b*

rm -r b.exe

echo ""
echo "copy c is prioritary on copy of c.exe: cp c copy_c"
cp c copy_c
echo "* cat copy_c"
cat copy_c

rm -r copy_c

echo ""
echo "copy 'a.exe' in an existing folder: cp a dir_a"
cp a dir_a
echo "* ls dir_a/*"
ls dir_a/*
echo "* cat dir_a/a.exe"
cat dir_a/a.exe

rm -r dir_a/a.exe
