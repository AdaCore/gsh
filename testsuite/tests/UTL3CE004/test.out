when 'a' does not exist but 'a.exe' does: cp a copy_a
* ls copy_a.exe
copy_a.exe
* cat copy_a.exe
content of a.exe

copy on side of an existing file: cp a b
* ls b*
b
b.exe
* cat b*
content of b - must not be overwritten when `cp a b` with 'a' being actually a.exe
content of a.exe

copy c is prioritary on copy of c.exe: cp c copy_c
* cat copy_c
content of c

copy 'a.exe' in an existing folder: cp a dir_a
* ls dir_a/*
dir_a/a.exe
* cat dir_a/a.exe
content of a.exe
